yttc/plugin/tosca.py

```diff
@@ -74,53 +66,79 @@ def emit_yaml(ctx, modules, fd):
         else:
             prefix = "{}".format(module.i_prefix)
             xmlns[prefix] = namespace
+        rev = module.search_one('revision')
+        if rev:
+            revision = "{}".format(rev.arg)
+        else:
+            revision = 'unknown'
+        uses_modules.update({"{}".format(module.arg): revision})
     if not main_module:
         message = "Can't find main module."
         fd.write(message)
         return False, message
-    output['node_types'][main_module] = {'properties':
-                                         {'edit-config':
-                                          {'type': 'edit-config',
-                                           'required': False},
-                                          'metadata':
-                                          {'default':
-                                           {'xmlns': xmlns}}}}
-    output['data_types']['edit-config'] = {'properties':
-                                           {'target': {'default': 'running'},
-                                            'config': {'type': 'config'}}}
+    dsdl = _get_dsdl_representation(ctx, modules)
+    output = {
+        'data_types': {
+            'config': {
+                'properties': {
+                }},
+            'rpc': {
+                'properties': {
+                }}},
+        'node_types': {
+            main_module: {
+                'derived_from': 'cloudify.netconf.nodes.xml_rpc',
```

Comment:
This is what I referred to with the "xml_rpc"
It is better not to have this hardcoded inside the tosca.py, but load this from an external file which can be changed by specifying a different flag.